temoa/utilities/visualizer.py

```
@@ -0,0 +1,311 @@
+    if vis_options:
+        deep_merge_dicts(vis_options, current_options)
+
+    try:
+        template_dir = Path(__file__).parent / 'network_vis_templates'
+        html_template = (template_dir / 'graph_template.html').read_text(encoding='utf-8')
+        css_template = (template_dir / 'graph_styles.css').read_text(encoding='utf-8')
+        js_template = (template_dir / 'graph_script.js').read_text(encoding='utf-8')
+    except FileNotFoundError as e:
+        logger.error(
+            f"Template file not found: {e}. Ensure the 'templates' directory exists next to visualizer.py."
+        )
```

Ruff (0.14.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)